internal/namespaces/instance/v1/custom_server_create.go

```diff
@@ -453,8 +470,32 @@ func buildVolumeTemplateFromUUID(api *instance.API, zone scw.Zone, volumeUUID st
 	}, nil
 }
 
+func validateImageServerTypeCompatibility(getImageResponse *instance.GetImageResponse, serverType *instance.ServerType, CommercialType string) error {
```

Comment:
We don't need the whole `*instance.GetImageResponse` here:
`func validateImageServerTypeCompatibility(image *instance.Image, serverType *instance.ServerType, commercialType string) error`

(`CommercialType` => `commercialType`)